tests/gpflow/likelihoods/test_likelihoods.py

```diff
@@ -379,3 +382,107 @@ def test_robust_max_multiclass_eps_k1_changes(num_classes, initial_epsilon, new_
     expected_eps_k2 = new_epsilon / (num_classes - 1.0)
     actual_eps_k2 = likelihood.eps_k1
     assert_allclose(expected_eps_k2, actual_eps_k2)
+
+
+@pytest.mark.parametrize("Y_list", [[tf.random.normal((i, 2)) for i in range(3, 6)]])
+@pytest.mark.parametrize("F_list", [[tf.random.normal((i, 2)) for i in range(3, 6)]])
+@pytest.mark.parametrize("Fvar_list", [[tf.exp(tf.random.normal((i, 2))) for i in range(3, 6)]])
+@pytest.mark.parametrize("Y_label", [[tf.ones((i, 2)) * (i - 3.0) for i in range(3, 6)]])
+def test_switched_likelihood_log_prob(Y_list, F_list, Fvar_list, Y_label):
+    """
+    SwitchedLikelihood is separately tested here.
+    Here, we make sure the partition-stitch works fine.
+    """
+    Y_perm = list(range(3 + 4 + 5))
+    np.random.shuffle(Y_perm)
+    # shuffle the original data
+    Y_sw = np.hstack([np.concatenate(Y_list), np.concatenate(Y_label)])[Y_perm, :3]
+    F_sw = np.concatenate(F_list)[Y_perm, :]
+    likelihoods = [Gaussian()] * 3
+    for lik in likelihoods:
+        lik.variance = np.exp(np.random.randn(1)).squeeze().astype(np.float32)
+    switched_likelihood = SwitchedLikelihood(likelihoods)
+
+    switched_results = switched_likelihood.log_prob(F_sw, Y_sw)
+    results = [lik.log_prob(f, y) for lik, y, f in zip(likelihoods, Y_list, F_list)]
+
+    assert_allclose(switched_results, np.concatenate(results)[Y_perm])
+
+
+@pytest.mark.parametrize("Y_list", [[tf.random.normal((i, 2)) for i in range(3, 6)]])
+@pytest.mark.parametrize("F_list", [[tf.random.normal((i, 2)) for i in range(3, 6)]])
+@pytest.mark.parametrize("Fvar_list", [[tf.exp(tf.random.normal((i, 2))) for i in range(3, 6)]])
+@pytest.mark.parametrize("Y_label", [[tf.ones((i, 2)) * (i - 3.0) for i in range(3, 6)]])
+def test_switched_likelihood_predict_density(Y_list, F_list, Fvar_list, Y_label):
+    Y_perm = list(range(3 + 4 + 5))
```

Comment:
the SwitchedLikelihood tests have moved to a new file - can you please move them over there ?